service/health/types.go

```diff
@@ -68,30 +68,54 @@ func NewClusterStatus(cluster v1alpha1.StatusCluster, nodes []v1.Node) ClusterSt
 	}
 }
 
-func NewNodeStatus(node v1.Node) NodeStatus {
+func NewNodesStatus(nodes []v1.Node, pods []v1.Pod) []NodeStatus {
+	result := []NodeStatus{}
+	for _, node := range nodes {
+		result = append(result, NewNodeStatus(node, filterNodePods(pods, node.Name)))
```

Comment:
That's a nit but I think `filterNodePods` should be done in `NewNodeStatus`. Also do those constructors need to be exposed?